### PR TITLE
Bump core to 3.1.0 and add h3Distance

### PR DESCRIPTION
**Pull request by nrabinowitz**
* Bumps core library to 3.1.0
* Adds binding for h3Distance, with basic test
* Updates CHANGELOG

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,11 @@
 All notable changes to this project will be documented in this file. This library adheres to a versioning policy described in [the README](./README.md#versioning). The public API of this library consists of the functions exported in [h3core.js](./lib/h3core.js).
 
 ## [Unreleased]
+### Added
```
Comment (isaacbrodsky) [inline CHANGELOG.md]: Are we adding the PR numbers later or in the PR that adds the Changelog line?

File: test/h3core.spec.js [hunk 2]
```diff
@@ -1128,6 +1128,21 @@ test('getH3UnidirectionalEdgeBoundary - 10-vertex pentagon', assert => {
     assert.end();
 });
 
+test('h3Distance', assert => {
```
Comment (isaacbrodsky) [inline test/h3core.spec.js]: We should have a test for the case where a negative distance (failure) is returned.

File: CHANGELOG.md (hunk 1 above)
Comment (nrabinowitz) [reply CHANGELOG.md]: Oh, that's a good point, will add.

File: test/h3core.spec.js (hunk 2 above)
Comment (nrabinowitz) [reply test/h3core.spec.js]: Will add.

File: CHANGELOG.md (hunk 1 above)
Comment (nrabinowitz) [reply CHANGELOG.md]: Added.

File: test/h3core.spec.js (hunk 2 above)
Comment (nrabinowitz) [reply test/h3core.spec.js]: Added.

**Pull request merged**